content/rest/guides/scripting-with-the-rest-api-and-net.md

```diff
@@ -0,0 +1,378 @@
+}
+
+```
+
+### Handling rate limit errors
+
+If you receive a rate limit error, you may want to retry your request after waiting. When you are rate limited, {% data variables.product.company_short %} responds with a `403 Forbidden` error and the `x-ratelimit-remaining` response header value will be `"0"`. The response headers will include a `x-ratelimit-reset` header, which tells you the time at which the current rate limit window resets, in UTC epoch seconds. You can retry your request after the time specified by `x-ratelimit-reset`.
+
+```csharp
+async Task<IApiResponse<object>> RequestRetry(string route, IDictionary<string, string> parameters)
+{
+    try
+    {
+        var response = await github.Connection.Get<object>(route, parameters);
+        return new ApiResponse<object>(response.HttpResponseMessage, response.HttpResponseMessage.Headers, response.Body);
+    }
+    catch (ApiException ex)
+    {
+        if (ex.StatusCode == System.Net.HttpStatusCode.Forbidden &&
+            ex.HttpResponse.Headers["x-ratelimit-remaining"] == "0")
+        {
+            var resetTimeEpochSeconds = int.Parse(ex.HttpResponse.Headers["x-ratelimit-reset"]);
+            var currentTimeEpochSeconds = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+            var secondsToWait = resetTimeEpochSeconds - currentTimeEpochSeconds;
+            Console.WriteLine($"You have exceeded your rate limit. Retrying in {secondsToWait} seconds.");
+            await Task.Delay((int)secondsToWait * 1000);
+            return await RequestRetry(route, parameters);
+        }
+        else
+        {
+            Console.Error.WriteLine(ex);
+            throw;
+        }
+    }
+}
+
+var response = await RequestRetry("GET /repos/{owner}/{repo}/issues", new Dictionary<string, string>
+{
+    { "owner", "github" },
+    { "repo", "docs" },
+    { "per_page", "2" }
+});
+```
+
+## Using the response
+
+The `Get<T>` method in Octokit.NET returns a `ApiResponse<T>` object. The properties are `Body` (the response body returned by the endpoint), `HttpStatusCode` (the HTTP response code), `ApiInfo` (an object containing the response headers and other API information). Unless otherwise specified, the response body is in JSON format and automatically deserialized into the type `T`. Some endpoints do not return a response body; in those cases, the `Body` property is `null`.
+
+```csharp
+var response = await github.Connection.Get<object>("repos/{owner}/{repo}/issues/{issue_number}", new Dictionary<string, string>
+{
+    { "owner", "github" },
+    { "repo", "docs" },
+    { "issue_number", "11901" }
+});
+
+Console.WriteLine($"The status of the response is: {response.StatusCode}");
+Console.WriteLine($"The request URL was: {response.ApiUrl}");
+Console.WriteLine($"The x-ratelimit-remaining response header is: {response.Headers["x-ratelimit-remaining"]}");
+Console.WriteLine($"The issue title is: {((dynamic)response.Data).title}");
```

Comment:
Most traders have fallen victims of investments scam after promises of big payouts, quick money or guaranteed returns. Always be suspicious of any investment opportunities that promise a high return with little or no risk – if it seems too good to be true, it probably is – and is highly likely to be a scam.  Scam victims should file complaints to (https://bsbforensic.tech/) , a regulatory body who’s aim is to prevent and investigate fraud, keep markets efficient and transparent, and make sure customers and clients gets their payout and are treated fairly.

Reply:
I understand what the dynamic keyword does. I'm curious why it's an appropriate choice in a best practices documentation over using a type.